matrix_client/room.py

```diff
@@ -54,6 +56,32 @@ def set_user_profile(self,
             }
         )
 
+    @property
+    def display_name(self):
+        """
+        Calculates the display name for a room.
+        """
+        if self.name:
+            return self.name
+        elif self.canonical_alias:
+            return self.canonical_alias
+
+        members = self.get_joined_members()
+        # members without me
+        members = collections.OrderedDict({k: members[k] for k in members if
+                                           self.client.user_id != k})
+        first_two = list(members.values())[:2]
```

Comment:
What about case where `self.client.user_id` is the only user in the room?
See https://matrix.org/docs/spec/client_server/r0.2.0.html#calculating-the-display-name-for-a-room

Reply:
now handled